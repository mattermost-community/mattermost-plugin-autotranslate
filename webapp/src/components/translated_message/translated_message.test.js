import '@testing-library/jest-dom';
import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';

import TranslatedMessage from './translated_message';

test('should not render any message', async () => {
    const hideTranslatedMessage = jest.fn();
    const {container, rerender} = render(
        <TranslatedMessage
            activated={false}
            hideTranslatedMessage={hideTranslatedMessage}
        />,
    );
    expect(container).toMatchInlineSnapshot('<div />');

    rerender(
        <TranslatedMessage
            translation={{show: false}}
            activated={false}
            hideTranslatedMessage={hideTranslatedMessage}
        />,
    );
    expect(container).toMatchSnapshot();
});

test('should render error message', async () => {
    const hideTranslatedMessage = jest.fn();
    const onHeightChange = jest.fn();
    const translation = {
        post_id: 'post_id_1',
        show: true,
        errorMessage: 'Failed to get translation',
    };

    const {container} = render(
        <TranslatedMessage
            translation={translation}
            activated={true}
            hideTranslatedMessage={hideTranslatedMessage}
            onHeightChange={onHeightChange}
        />,
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText(translation.errorMessage)).toBeInTheDocument();

    fireEvent.click(screen.getByText(/close/i));
    expect(hideTranslatedMessage).toHaveBeenCalledTimes(1);
    expect(hideTranslatedMessage).toHaveBeenCalledWith(translation.post_id);
    expect(onHeightChange).toHaveBeenCalledTimes(1);
    expect(onHeightChange).toHaveBeenCalledWith(1);
});

test('should render translated message', async () => {
    const hideTranslatedMessage = jest.fn();
    const onHeightChange = jest.fn();
    const translation = {
        post_id: 'post_id_2',
        show: true,
        translated_text: 'Hello world',
    };

    const {container} = render(
        <TranslatedMessage
            translation={translation}
            activated={true}
            hideTranslatedMessage={hideTranslatedMessage}
            onHeightChange={onHeightChange}
        />,
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText(translation.translated_text)).toBeInTheDocument();
    expect(screen.getByText(/See translation/i)).toBeInTheDocument();

    fireEvent.click(screen.getByText(/close/i));
    expect(hideTranslatedMessage).toHaveBeenCalledTimes(1);
    expect(hideTranslatedMessage).toHaveBeenCalledWith(translation.post_id);
    expect(onHeightChange).toHaveBeenCalledTimes(1);
    expect(onHeightChange).toHaveBeenCalledWith(1);
});
