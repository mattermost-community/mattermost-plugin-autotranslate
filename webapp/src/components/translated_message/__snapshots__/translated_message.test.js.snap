// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render any message 2`] = `<div />`;

exports[`should render error message 1`] = `
<div>
  <p>
    <i
      class="icon fa fa-language"
    />
    <span
      style="color: red;"
    >
        Failed to get translation  
    </span>
    <a>
      (close)
    </a>
  </p>
</div>
`;

exports[`should render translated message 1`] = `
<div>
  <p>
    <i
      class="icon fa fa-language"
    />
    <span>
        See translation:

    </span>
    <span>
      Hello world  
    </span>
    <a>
      (close)
    </a>
  </p>
</div>
`;
